pkg/library/flatten/flatten.go

```diff
@@ -202,5 +204,31 @@ func resolvePluginComponentById(
 		return &pluginDWT.Spec, pluginDWT.Labels, nil
 	}
 
-	return nil, nil, fmt.Errorf("non-internal plugins not supported")
+	pluginURL, err := url.Parse(plugin.RegistryUrl)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to parse registry URL for plugin %s: %w", name, err)
+	}
+	pluginURL.Path = path.Join(pluginURL.Path, "plugins", plugin.Id)
+
+	dwt, labels, err := network.FetchDevWorkspaceTemplate(pluginURL.String(), tools.HttpClient)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to resolve plugin %s from registry %s: %w", name, plugin.RegistryUrl, err)
+	}
+	return dwt, labels, nil
+}
+
+func resolvePluginComponentByURI(
+	name string,
+	plugin *devworkspace.PluginComponent,
+	tools ResolverTools) (resolvedPlugin *devworkspace.DevWorkspaceTemplateSpec, pluginLabels map[string]string, err error) {
+
+	if plugin.Uri == "" {
```

Comment:
Can this piece be hit? It looks like on line ~135 ish you have `case plugin.Uri != ""`

Reply:
You're right, it can't -- I suppose it's moreso paranoia on my part from constantly having to check `component.Plugin != nil` :smile: 

I'll remove the check (this also increases our test coverage :wink: !)